src/qmapshack/helpers/CMapIconSizesSetup.cpp

```diff
@@ -0,0 +1,83 @@
+/**********************************************************************************************
+    Copyright (C) 2014 Oliver Eichler <oliver.eichler@gmx.de>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+**********************************************************************************************/
+
+#include "helpers/CMapIconSizesSetup.h"
+
+#include <QPushButton>
+#include <QSlider>
+
+#include "helpers/CSettings.h"
+#include "helpers/CWptIconManager.h"
+#include "poi/IPoiFile.h"
+
+CMapIconSizesSetup::CMapIconSizesSetup(QWidget* parent)
+    : QDialog(parent), imageWpt("://icons/48x48/Waypoint.png"), imagePoi("://icons/48x48/Poi.png") {
+  setupUi(this);
+
+  int labelSize = qMax(sliderWpt->maximum(), sliderPoi->maximum());
```

Comment:
Just a minor detail: The label is a styled panel with a frame of size 1px. Therefore the inner area to display stuff is 46x46 instead of 48x48. This is why the icons seem to get a bit larger than the panel. Therefor labelSize should be +2 whatever qMax passes back.